### PR TITLE
add retry with exponential backoff for delegate clients with max retry configurable by env, disabled by default

### PR DESCRIPTION
**Pull request by k4leung4**
add retry with exponential backoff for delegate clients with max retry configurable by env, disabled by default

### CODE REVIEW

File: pkg/options/options.go [hunk 1]
```diff
@@ -29,6 +31,7 @@ var env struct {
 	EnableClientHandlingTimeHistogram      bool `envconfig:"ENABLE_CLIENT_HANDLING_TIME_HISTOGRAM" default:"true"`
 	EnableClientStreamReceiveTimeHistogram bool `envconfig:"ENABLE_CLIENT_STREAM_RECEIVE_TIME_HISTOGRAM" default:"true"`
 	EnableClientStreamSendTimeHistogram    bool `envconfig:"ENABLE_CLIENT_STREAM_SEND_TIME_HISTOGRAM" default:"true"`
+	GrpcClientMaxRetry                     int  `envconfig:"GRPC_CLIENT_MAX_RETRY" default:"5"`
```
Comment (tcnghia) [inline pkg/options/options.go:34]: I am not familiar with customization option for envconfig, but can a CLI specify its own config here?

Comment (tcnghia) [reply pkg/options/options.go:34]: Also, IIUC @mattmoor suggests default to be 0 here (WithMax(0) will disable retry according to the doc)

Comment (k4leung4) [reply pkg/options/options.go]: if they can set an env var, they can set this.

not sure how that works on windows in cli though

Comment (k4leung4) [reply pkg/options/options.go]: changed to default to 0

**Review by tcnghia — APPROVED** [review]

**Pull request merged**